### ISSUE TITLE
[Feature] - Support Github actions yaml

### ISSUE THREAD

**Issue by dinbtechit** [microsoft/appcenter#2493]
**Describe the solution you'd like**
The build functional in App Center is extremely limited. Is it possible to support github action yaml files in App center?

**Describe alternatives you've considered**
Using github actions for building.

**Comment by dinbtechit** [comment]
Hi @MikhailSuendukov  - That one is for Azure DevOps Yaml. It's not the same as GitHub Actions Yaml. I would say using Github Action Yaml would cover a wider range of users as Github has more user base than Azure DevOps. 

Please advise if you still think this is a duplicate of #1905

**Comment by MikhailSuendukov** [comment]
I meant, did I understand correctly that you were talking about similar behavior for Github actions?

**Comment by dinbtechit** [comment]
That is correct. Similar behavior but with Github actions.

**Comment by dinbtechit** [comment]
@DmitriyKirakosyan  -  Hmm. I dont think I explained this correctly. Let me try to rephrase my request. So I am looking for an environment/machine where I can build my apps. A side note - we use  Self Hosted GitLab as our source control.

Since we are already using App Center for some of our apps. We thought we could use App Center for our React Native app as well.  But for some reason, both ios and the android build fails on the App Center. I am not sure what commands the App Center is using to do the builds but it doesn't seem to work for us. (See screenshots below)

On the contrary, the Android build works perfectly fine on the GitLab ci/cd (unfortunately we don't have a macOS runner to build ios on our GitLab CI. But I was able successfully build them on my local MacBook pro).  So it would be nice if we could modify the build commands for our project. And, rather than coming up with some custom format, I thought it would be better to use the widely adopted github actions yaml file format for App Center as well.

Hopefully, this explains.


Android Build Error:
![image](https://user-images.githubusercontent.com/17984781/205937784-8f882337-8e01-4dbf-ad52-2b21f5ea0b2f.png)


IOS build Error:
![image](https://user-images.githubusercontent.com/17984781/205937386-99f80e4a-7aa4-45b5-a974-cbadd5396449.png)
